lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -4932,6 +4930,41 @@ namespace Js
         }
     }
 
+    bool JavascriptOperators::CanShortcutOnUnknownPropertyName(RecyclableObject *instance)
+    {
+        if (!CanShortcutInstanceOnUnknownPropertyName(instance))
+        {
+            return false;
+        }
+        return CanShortcutPrototypeChainOnUnknownPropertyName(instance->GetPrototype());
+    }
+
+    bool JavascriptOperators::CanShortcutInstanceOnUnknownPropertyName(RecyclableObject *instance)
+    {
+        if (!instance->HasDeferredTypeHandler())
```

Comment:
Actually, Proxy is the only case where it doesn't matter what kind of type handler we have. I'll refactor this.